### PR TITLE
Bjsvedin unit tests

### PR DESCRIPTION
**Pull request by bjsvedin**
I have fixed up the drawCard function and have created an extensive unit test for it. It properly checks if the playerID matches the currentPlayers ID, it checks against empty drawPiles and discardPiles, though if all are empty we need something more complicated then just throwing another deck on the pile. It then swaps the cards around, saves the game and returns.

### CODE REVIEW

File: server/uno.go [hunk 1]
```diff
@@ -222,34 +212,44 @@ func drawCard(gameID string, playerID string) (*model.Game, error) {
...
-					game.DiscardPile = game.DiscardPile[:0]
-				}
+	// We get the current player from the game
+	player := &game.Players[game.CurrentPlayer]
+	//We then check if the player attempting to play a card is the current player
+	if player.ID == playerID {
+		// We check if the draw pile has available cards
+		if len(game.DrawPile) == 0 {
+			// we check that the discard pile has cards to reshuffle
+			if len(game.DiscardPile) <= 1 {
```
Comment (marwtki) [inline server/uno.go:222]: Nice work thinking of all these scenarios, like when there is nothing in the drawpile or discard pile

File: server/uno_test.go [hunk 2]
```diff
@@ -8,63 +8,118 @@ import (
...
+
+	// Create a player to simulate one playing but its not their turn and attempt to play
+	otherPlayer := model.Player{ID: " id 2 ", Name: "Name 2", Cards: []model.Card{}}
 
-	// game, player := setupGameWithPlayer(database)
+	_, err = drawCard(game.ID, otherPlayer.ID)
 
-	// game, err := drawCard(game.ID, player.ID)
+	// Assert that we got an error from the draw card.
+	assert.NotNil(t, err, "Player played out of turn. Please make sure only the player whoes turn it is can play.")
```
Comment (marwtki) [inline server/uno_test.go:121]: What's the third argument here? Does `assert.NotNil` just check that `err != nil` and if so what's the use of the string? I don't think it matches what the error will be

File: server/uno_test.go [hunk 3]
```diff
@@ -8,63 +8,118 @@ import (
...
+	//Assert no error, player has 2 cards from both draw tests,
+	// draw is missing three from two draws and one in discard
+	// discard has one last remaining card.
+	// Assert last card in discard is actually to proper last card
+	assert.Nil(t, err, "Failed to draw card.")
+	assert.Equal(t, 2, len(player.Cards))
+	assert.Equal(t, 105, len(game.DrawPile))
+	assert.Equal(t, 1, len(game.DiscardPile))
+	assert.Equal(t, lastCard.Color, game.DiscardPile[0].Color)
+	assert.Equal(t, lastCard.Value, game.DiscardPile[0].Value)
```
Comment (marwtki) [inline server/uno_test.go:92]: Great job with the thorough testing!

File: server/uno_test.go (hunk 2 above)
Comment (bjsvedin) [reply server/uno_test.go:121]: The String is what you get in the console if the test case fails. Just before this i simulate a player trying to draw a card out of turn. I create a bogus player with a bogus id and pass them into Draw Card. We should get an error message so i check that the error exists. It should not allow them to draw a card. Only the players who's turn it is should be able to draw a card.  I should probably assert a couple more items around both the bogus player and the game. Also it should say "Player 'drew' a card out of turn".

Comment (bjsvedin) [reply server/uno_test.go:121]: I should check that the message from the error is also the one i expect. There are two other places in drawCard that an error can be returned.

**Pull request merged**